crates/rust-analyzer/src/handlers/notification.rs

```diff
@@ -352,20 +353,30 @@ fn run_flycheck(state: &mut GlobalState, vfs_path: VfsPath) -> bool {
                     | project_model::ProjectWorkspaceKind::DetachedFile {
                         cargo: Some((cargo, _, _)),
                         ..
-                    } => cargo.packages().find_map(|pkg| {
-                        let has_target_with_root = cargo[pkg]
-                            .targets
-                            .iter()
-                            .any(|&it| crate_root_paths.contains(&cargo[it].root.as_path()));
-                        has_target_with_root.then(|| cargo[pkg].name.clone())
-                    }),
+                    } => {
+                        // Iterate crate_root_paths first because it is in topological
+                        // order, and we can therefore find the actual crate your saved
```

Comment:
Ah, I believe this was true when I first wrote it, but this appears to be no longer the case.

Reply:
Now the reverse deps function is topological.